spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java

```diff
@@ -127,7 +129,7 @@ public static void configureServerWithArmeriaSettings(
         if (needsManagementPort && managementServerPort != null) {
             internalPortsBuilder.add(managementServerPort);
         }
-        final List<Port> internalPorts = internalPortsBuilder.build();
+        final List<Port> internalPorts = dedupPorts(internalPortsBuilder.build());
         configurePorts(server, settings.getPorts());
```

Comment:
How about also calling dedup here?
```suggestion
        configurePorts(server, dedupPorts(settings.getPorts()));
```